web/modules/custom/dpl_consumers/src/Services/ConsumerHandler.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\dpl_consumers\Services;
+
+use Drupal\dpl_consumers\Consumer;
+use Drupal\dpl_consumers\ConsumerRole;
+use Drupal\dpl_consumers\ConsumerUser;
+use Psr\Log\LoggerInterface;
+
+/**
+ * DPL consumer handler.
+ *
+ * This class is responsible for creating consumers and users.
+ */
+class ConsumerHandler {
```

Comment:
I'm not fond of this class. What is a "ConsumerHandler" anyway? 

On one hand it's a composition of consumer, user and role. On the other hand `setComponents` allows for setting new consumer, user and role.

In my mind it should either be a "handler", state free, and having `create` and `delete` just take the consumer, user and role as arguments.

Or it should be an object representing *this* combination (`ConsumerSetup` perhaps?), in which case it should take consumer, user and role as constructor arguments (and then you'd need a factory in order to get the container to inject the logger).